plugin/lib/class-php-compatibility-checker.php

```diff
@@ -102,73 +98,144 @@ public function init() {
 		// Handle activation notice.
 		// register_activation_hook( __FILE__, array( $instance, 'set_activation_notice_flag' ) );
 		// add_action( 'admin_notices', array( $instance, 'maybe_show_activation_notice' ) );
+	}
 
-		if ( is_admin() ) {
-			// not sure yet.
-		}
+	/**
+	 * Check if a plugin should be excluded from scans.
+	 *
+	 * @param array $plugin_data Plugin data.
+	 * @see https://developer.wordpress.org/reference/functions/get_plugin_data/
+	 * @return boolean
+	 */
+	public function exclude_plugin( $plugin_data ) {
+		$excluded_plugins = array( 'PHP Compatibility Checker', 'Hello Dolly' );
+
+		/**
+		 * Filter which plugins should be excluded from scans.
+		 *
+		 * This will exclude based on the plugin name, not the plugin slug.
+		 *
+		 * @param string[] $excluded_plugins Plugins we want to exclude.
+		 */
+		$excluded_plugins = apply_filters( 'phpcompat_excluded_plugins', $excluded_plugins );
+
+		return in_array( $plugin_data['Name'], $excluded_plugins, true );
 	}
 
 	/**
-	 * Generates a list of directories to scan and populate the queue.
+	 * Get all plugins that should be scanned.
 	 *
-	 * @since  1.0.0
+	 * @return array
 	 */
-	public function generate_directory_list() {
+	public function get_plugins_to_scan() {
 		if ( ! function_exists( 'get_plugins' ) ) {
-
+			// phpcs:ignore PEAR.Files.IncludingFile.UseIncludeOnce -- strict requirement
 			require_once ABSPATH . 'wp-admin/includes/plugin.php';
 		}
 
-		$plugin_base = dirname( $this->base ) . DIRECTORY_SEPARATOR;
-		$all_plugins = get_plugins();
-
-		$update_plugins = get_site_transient( 'update_plugins' );
-		//$slug_name[] = plugin_basename();
+		$plugins = get_plugins();
 
-		foreach ( $all_plugins as $k => $v ) {
-			// Exclude some plugins.
-			if ( 'PHP Compatibility Checker' === $v['Name'] ) {
-				continue;
-			}
-			if ( 'Hello Dolly' === $v['Name'] ) {
-				continue;
+		// Exclude some plugins.
+		$plugins = array_filter(
+			$plugins,
+			function ( $plugin_data ) {
+				return ! $this->exclude_plugin( $plugin_data );
 			}
+		);
```

Comment:
I know this is getting into the territory of micro-improvements but wondering if we should move the exclude filter into here. Otherwise that filter needs to be run for each individual plugin, which is minor but not the best for performance. And while people could use the `phpcompat_plugins_to_scan` filter to modify which plugins are scanned, I like having this exclusion filter as well since to me it's more clear on how to use it.

Maybe something like this (which would allow us to get rid of the `exclude_plugin` method all together):
```suggestion
                /**
		 * Filter which plugins should be excluded from scans.
		 *
		 * This will exclude based on the plugin name, not the plugin slug.
		 *
		 * @param string[] $excluded_plugins Plugins we want to exclude.
		 */
		$excluded_plugins = apply_filters( 'phpcompat_excluded_plugins', array( 'PHP Compatibility Checker', 'Hello Dolly' ) );

		// Exclude some plugins.
		$plugins = array_filter(
			$plugins,
			function ( $plugin_data ) use ( $excluded_plugins ) {
				return in_array( $plugin_data['Name'], $excluded_plugins, true );
			}
		);
```